packages/astro/src/core/routing/manifest/create.ts

```diff
@@ -899,3 +893,17 @@ function computeRoutePriority(config: AstroConfig): RoutePriorityOverride {
 	}
 	return 'legacy';
 }
+
+function joinSegments(segments: RoutePart[][]): string {
+	const arr = segments.map(function d(segment): string {
+		const [{ dynamic, content }] = segment;
+
+		return segment.length > 1
+			? segment.map((s) => d([s])).join('')
+			: dynamic
+				? `[${content}]`
+				: content;
```

Comment:
Spread segments should be `[...${content}]`

Reply:
`content` of rest parameter is `...xxx`,i'm not entirely certain whether it requires separate handling.